llvm/lib/Transforms/Utils/SCCPSolver.cpp

```diff
@@ -109,14 +101,14 @@ static bool refineInstruction(SCCPSolver &Solver,
                               Instruction &Inst) {
   bool Changed = false;
   auto GetRange = [&Solver, &InsertedValues](Value *Op) {
-    if (auto *Const = dyn_cast<ConstantInt>(Op))
-      return ConstantRange(Const->getValue());
-    if (isa<Constant>(Op) || InsertedValues.contains(Op)) {
+    if (auto *Const = dyn_cast<Constant>(Op))
```

Comment:
Do we need to guard against `ConstantExpr` case?

Reply:
See https://github.com/llvm/llvm-project/commit/d177a94fbdc12c82e06dc1c2dc7500c3ce399291. `ConstantExpr` should be handled to support splat vector constants.